File: Herança - Subclasses de Personagem/aula4_heranca_PY/Guerreiro.py
from Personagem import Personagem

class Guerreiro(Personagem):
    def __init__(self, nome, vida, nivel):
        super().__init__()
        self.nome = nome
        self.vida = vida
        self.nivel = nivel
        self._defesa = 5
    @property
    def defesa(self):
        return self._defesa
    @defesa.setter
    def defesa(self, valor):
        if valor < 0:
            print("Defesa inválida")
        else:
            self._defesa = valor

    def receber_dano(self, dano):
        super().receber_dano(max(0, dano - self._defesa))

    def ficha(self):
       return super().ficha() + "\nDefesa: " + str(self._defesa)

File: Exceções - IF Quest/java/excecoesifquest/Guerreiro.java
package excecoesifquest;

public class Guerreiro extends Personagem{

        private int defesa;

public Guerreiro(String nome, int vida, int nivel){
    super(nome, vida, nivel);
    defesa = 5;
}
public int getDefesa(){
    return defesa;
}
public void setDefesa(int defesa){
    if (defesa < 0){
        throw new IllegalArgumentException("A defesa não pode ser negativa: " + defesa);
    } else {
        this.defesa = defesa;
    }
}
@Override
public void receberDano(int dano){
    super.receberDano(Math.max(0, dano - defesa));
}
@Override
public String ficha(){
    return super.ficha() + "\nDefesa: " + defesa;
}

@Override
public String habilidade(){
    return "golpe poderoso";
}
}